### PR TITLE
Tutorial - Evaluate an LLM's prompt completions

### PR DESCRIPTION
**Pull request by haywoodsloan**
## Summary

Add the tutorial doc: *Tutorial: Evaluate an LLM's prompt completions*

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/ai/tutorials/llm-eval.md](https://github.com/dotnet/docs/blob/1023daad03ee3da294c5de91041762fb83cdc0f2/docs/ai/tutorials/llm-eval.md) | [Tutorial: Evaluate an LLM's prompt completions](https://review.learn.microsoft.com/en-us/dotnet/ai/tutorials/llm-eval?branch=pr-en-us-40847) |

### CODE REVIEW

**Comment by haywoodsloan** [comment]
@elbruno ready for review!

File: docs/ai/tutorial/llm-eval.md
Comment (elbruno) [inline docs/ai/tutorial/llm-eval.md:1]: In the [Related Content] section at the bottom, we may want to add a link to [Azure AI Studio - Evaluation of generative AI applications] 
https://learn.microsoft.com/en-us/azure/ai-studio/concepts/evaluation-approach-gen-ai as a relevant source for the custom metric implementation

Comment (haywoodsloan) [reply docs/ai/tutorial/llm-eval.md:1]: Updated!

**Comment by elbruno** [comment]
It's fine on my part.

File: docs/ai/tutorials/llm-eval.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+### Configure the model to generate test data
+
+The evaluation application is configured to use [the secrets set in the previous step](#configure-the-model-to-perform-evaluations) to connect to an Azure OpenAI model to generate test data. This configuration can be updated in the `KernelFactory.CreateKernelGenerateData` method:
+
+:::code language="csharp" source="./snippets/llm-eval/KernelFactoryExamples.cs" id="genKernel":::
+
+## 3 - Generate test data
+
+The evaluation application compares an LLM's output to "ground truth" answers, which are ideal question-answer pairs. At least 200 question-answer pairs are recommended for an evaluation.
```
Comment (gewarren) [inline docs/ai/tutorials/llm-eval.md:88]:
```suggestion
The evaluation application compares an LLM's output to "ground truth" answers, which are ideal question-answer pairs. It's recommended to have at least 200 question-answer pairs for an evaluation.
```

File: docs/ai/tutorials/llm-eval.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+You can use the evaluation application to generate an initial set of question-answer pairs. Then manually curate them, rewriting or removing any subpar answers.
```
Comment (gewarren) [inline docs/ai/tutorials/llm-eval.md:90]:
```suggestion
You can use the evaluation application to generate an initial set of question-answer pairs. Then manually curate them by rewriting or removing any subpar answers.
```

File: docs/ai/tutorials/llm-eval.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+* Be aware of the knowledge distribution so you effectively sample questions across the relevant knowledge space.
+* Once your application is live, continually sample real user questions (within accordance to your privacy policy) to make sure you're representing the kinds of questions that users are asking.
+
+1. From the `ai-samples/src/llm-eval/LLMEval.Test` directory, run the following command:
+
+    ```dotnetcli
+    dotnet run .
+    ```
+
+1. Select **Generate QAs associated to a topic, and export to json**, then press **Enter**.
```
Comment (gewarren) [inline docs/ai/tutorials/llm-eval.md:104]:
```suggestion
1. Select **Generate QAs associated to a topic, and export to json**, then press <kbd>Enter</kbd>.
```

File: docs/ai/tutorials/llm-eval.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+1. Enter the number of question-answer pairs to be generated and their topic.
+
+    :::image type="content" source="../media/llm-eval/eval-app-gen-input.png" lightbox="../media/llm-eval/eval-app-gen-input.png" alt-text="Number and topic inputs for question-answer generation with the Evaluation Application":::
+
+1. A preview of the generated question-answer pairs in JSON format is shown; enter the path of the file to save the JSON to.
+
+    :::image type="content" source="../media/llm-eval/eval-app-gen-output.png" lightbox="../media/llm-eval/eval-app-gen-output.png" alt-text="Output file input for question-answer generation with the Evaluation Application":::
+
+1. Review the output JSON, update or remove any incorrect or subpar answers.
```
Comment (gewarren) [inline docs/ai/tutorials/llm-eval.md:116]:
```suggestion
1. Review the output JSON, and update or remove any incorrect or subpar answers.
```

File: docs/ai/tutorials/llm-eval.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+## 4 - Perform an evaluation
+
+Once the question-answer pairs have been curated, the evaluation application can use them to evaluate the outputs of the test model.
```
Comment (gewarren) [inline docs/ai/tutorials/llm-eval.md:120]:
```suggestion
Once you've curated the question-answer pairs, the evaluation application can use them to evaluate the outputs of the test model.
```

File: docs/ai/tutorials/llm-eval.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+1. Copy the JSON file containing the question-answer pairs to `ai-samples/src/llm-eval/LLMEval.Test/assets/qa-02.json`.
+1. From the `ai-samples/src/llm-eval/LLMEval.Test` directory, run the following command:
+
+    ```dotnetcli
+    dotnet run .
+    ```
+
+1. Select **List of QAs from a file**, then press **Enter**.
```
Comment (gewarren) [inline docs/ai/tutorials/llm-eval.md:129]:
```suggestion
1. Select **List of QAs from a file**, then press <kbd>Enter</kbd>.
```

File: docs/ai/tutorials/llm-eval.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+In this tutorial, you evaluate the coherence, relevance, and groundedness of an LLM's prompt completions using Azure OpenAI and the Semantic Kernel SDK for .NET.
+
+:::image type="content" source="../media/llm-eval/eval-app.png" lightbox="../media/llm-eval/eval-app.png" alt-text="Main UI of the Evaluation Application":::
+
+In this tutorial, you learn how to:
+
+> [!div class="checklist"]
+>
+> * Clone and build the evaluation application
+> * Configure the models to test, generate test data, and perform evaluations
```
[the remark this answers is not in the document]
Comment (gewarren) [reply docs/ai/tutorials/llm-eval.md:23]: I found it a bit ambiguous as written. E.g. "configure the models to test [and configure the models to] generate test data" vs. "configure the models to test [and then] generate test data"

File: docs/ai/tutorials/llm-eval.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+1. Clone the repository [dotnet/ai-samples](https://github.com/dotnet/ai-samples)
+1. From a terminal or command prompt, navigate to the `ai-samples/src/llm-eval` directory.
+1. Build the evaluation application:
+
+    ```dotnetcli
+    dotnet build .
+    ```
+
+## 2 - Configure the models to test, perform evaluations, and generate test data
```
Comment (gewarren) [inline docs/ai/tutorials/llm-eval.md:47]:
```suggestion
## 2 - Configure the models
```

**Review by gewarren — APPROVED** [review]

**Review by elbruno — APPROVED** [review]
Add the LLM Eval to the TOC

**Pull request merged**